### PR TITLE
MariaDB Compatibility

### PR DESCRIPTION
**Pull request by alecc-bq**
### Summary

RDS-1139

### Description

- Added deregistration of mysql driver for mariadb driver tests
- Added prepared statement query accessor for mariadb
- Implemented mariadb driver compatibility for mysql using permitMysqlScheme parameter in Driver class and DataSource class

### CODE REVIEW

**Review by sergiyvamz — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java [hunk 1]
```diff
@@ -99,6 +99,11 @@ public Connection getConnection(String username, String password) throws SQLExce
       // If the url is set explicitly through setJdbcUrl or the connection properties.
       if (!isNullOrEmpty(this.jdbcUrl)
           || (!isNullOrEmpty(this.urlPropertyName) && !isNullOrEmpty(props.getProperty(this.urlPropertyName)))) {
+        if (!isNullOrEmpty(this.jdbcUrl)) {
```
Comment (karenc-bq) [inline wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java:102]: We can just use `StringUtils.isNullOrEmpty` instead, and remove this local `isNullOrEmpty` implementation.

File: wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java [hunk 2]
```diff
@@ -337,8 +342,8 @@ private void setJdbcUrlOrUrlProperty(Properties props) {
         && (!isNullOrEmpty(this.urlPropertyName) && !isNullOrEmpty(props.getProperty(this.urlPropertyName)))) {
       this.jdbcUrl = props.getProperty(this.urlPropertyName);
 
-      // If the jdbc url and the url property have both been set, use the provided jdbc url.
-    } else if ((!isNullOrEmpty(this.urlPropertyName) && !isNullOrEmpty(props.getProperty(this.urlPropertyName)))) {
+      // If the url property wasn't set, use the provided jdbc url.
+    } else if (!isNullOrEmpty(this.urlPropertyName)) {
```
Comment (karenc-bq) [inline wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java:346]: What if this.jdbcUrl is null? Will this result in a NullPointerException?

File: wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java [hunk 3]
```diff
@@ -114,4 +116,72 @@ public static void setPropertyOnTarget(
     }
     return copy;
   }
+
+  @SuppressWarnings("checkstyle:LocalVariableName")
+  public static @Nullable Properties parseProperties(String url, @Nullable Properties defaults) {
```
Comment (karenc-bq) [inline wrapper/src/main/java/software/amazon/jdbc/util/PropertyUtils.java:121]: Is this method necessary? We already have a `ConnectionUrlParser`

File: wrapper/src/test/java/integration/container/aurora/mysql/mariadbdriver/AuroraMysqlAwsIamIntegrationTest.java [hunk 4]
```diff
@@ -22,9 +22,11 @@
 import java.sql.SQLException;
 import java.util.Properties;
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import software.amazon.jdbc.PropertyDefinition;
 
+@Disabled
 public class AuroraMysqlAwsIamIntegrationTest extends MariadbAuroraMysqlBaseTest {
```
Comment (karenc-bq) [inline wrapper/src/test/java/integration/container/aurora/mysql/mariadbdriver/AuroraMysqlAwsIamIntegrationTest.java:30]: Should this be enabled?

File: wrapper/src/test/java/integration/container/standard/mysql/mysqldriver/MysqlStandardMysqlBaseTest.java [hunk 5]
```diff
@@ -33,7 +34,7 @@ public static void setUpMysql() throws SQLException, IOException, ClassNotFoundE
     STANDARD_USERNAME = System.getenv("STANDARD_MYSQL_USERNAME");
     STANDARD_PASSWORD = System.getenv("STANDARD_MYSQL_PASSWORD");
     setUp();
-    Class.forName("com.mysql.cj.jdbc.Driver");
+    DriverManager.registerDriver(new com.mysql.cj.jdbc.Driver());
```
Comment (karenc-bq) [inline wrapper/src/test/java/integration/container/standard/mysql/mysqldriver/MysqlStandardMysqlBaseTest.java:37]: What is wrong with the original `Class.forName`?

**Pull request closed without merging**